src/Core/Configurations/RuntimeConfigValidator.cs

```diff
@@ -162,12 +162,10 @@ public void ValidateAppInsightsTelemetryConnectionString(RuntimeConfig runtimeCo
         ValidateConfigProperties();
         ValidatePermissionsInConfig(runtimeConfig);
 
-        // If the ConfigValidationExceptions list doesn't contain a DataApiBuilderException with connection string error message,
-        // then only we run the metadata validation.
-        if (!ConfigValidationExceptions.Any(x => x.Message.Equals(DataApiBuilderException.CONNECTION_STRING_ERROR_MESSAGE)))
-        {
-            await ValidateEntitiesMetadata(runtimeConfig, loggerFactory);
-        }
+        _logger.LogInformation("Validating entity relationships.");
+        ValidateEntityRelationshipsInConfigWithoutDBMetadata(runtimeConfig);
+        await ValidateEntityRelationshipsInConfigWithGeneratedMetadata(runtimeConfig, loggerFactory);
```

Comment:
Ideas for shorter and more explicit naming: 
- `ValidateEntityRelationshipsWithDatabaseMetadata`
- `ValidateRelationshipsWithDatabaseMetadata`
- `ValidateRelationshipConfigWithDatabase`